include/jemalloc/internal/hpa.h

```diff
@@ -9,12 +9,40 @@
 typedef struct hpa_shard_nonderived_stats_s hpa_shard_nonderived_stats_t;
 struct hpa_shard_nonderived_stats_s {
 	/*
-	 * The number of times we've purged a hugepage.  Each eviction purges a
-	 * single hugepage.
+	 * The number of times we've fully purged a hugepage and evicted it from
+	 * the psset.
 	 *
-	 * Guarded by the grow mutex.
+	 * Guarded by grow_mtx.
```

Comment:
Looks this was removed later on.

Reply:
Just to answer anyways -- the grow pathways reuse previously mapped memory preferentially. We hold the grow mutex while touching it because we're adding it to the list of places to grow into.